connection.go

```diff
@@ -600,7 +614,14 @@ func (c *Connection) heartbeater(interval time.Duration, done chan *Error) {
 			// When reading, reset our side of the deadline, if we've negotiated one with
 			// a deadline that covers at least 2 server heartbeats
 			if interval > 0 {
-				conn.SetReadDeadline(time.Now().Add(maxServerHeartbeatsInFlight * interval))
+				if err := conn.SetReadDeadline(time.Now().Add(maxServerHeartbeatsInFlight * interval)); err != nil {
+					// TODO #78 what to do in error situation?
+					var op_error *net.OpError
+					if !errors.As(err, &op_error) {
```

Comment:
Would [errors.Is](https://pkg.go.dev/errors#Is) be better here given that variable `op_error` isn't used?

Reply:
In Go, use camel case instead of snake case.

Reply:
I think we'd need an instance of `net.OpError` for `errors.Is` to work. I'll keep it as `.As` and will use the value.